src/components/implementation/no_interface/boot_test/test_booter.c

```diff
@@ -0,0 +1,51 @@
+#include <stdio.h>
+#include <string.h>
+
+#include <cos_component.h>
+#include <cobj_format.h>
+#include <cos_kernel_api.h>
+#include "boot_deps.h"
+
+static void
+cos_llprint(char *s, int len)
+{ call_cap(PRINT_CAP_TEMP, (int)s, len, 0, 0); }
+
+
+int
+prints(char *s)
+{
+		int len = strlen(s);
+
+		cos_llprint(s, len);
+
+		return len;
+}
+
+int __attribute__((format(printf,1,2)))
```

Comment:
I think it would be best as a separate .h file for now as it will be easier to replace with an `components/interface/` based implementation later.